### ISSUE TITLE
Bun support

### ISSUE THREAD

**Issue by mheob** [wallabyjs/quokka#921]
### Issue description or question

[Bun](https://bun.sh/) released its first stable version a few weeks ago. It is also becoming more and more popular.

When I use Quokka with Bun I get this error:

```text
​​​​​Quokka PRO 'index.ts' (node: v20.9.0, TypeScript: v5.3.2)​​​​
 
Bun is not defined 
  ​​​​​at ​​​​​​​​readFile​​​ ​./src/utils/io.ts:2​
  ​​​​​at ​​​​​​​​readInput​​​ ​./src/utils/io.ts:8​
  ​​​​​at ​​​​​​src/day-04/index.ts:3:1
```

### Sample code

Bun provides its API. In my case, I use [`Bun.file("foo.txt")`](https://bun.sh/docs/api/file-io#reading-files-bun-file) to read the contents of a file.

```typescript
async function readFile(filepath: string) {
	const file = Bun.file(filepath);
	const text = await file.text();
	return text.trim();
}
```

This is just one example of a use.

### Sample repository link

The problem should be reproducible.\
But you can also try this repository for testing: https://github.com/mheob/advent-of-code-2023

### Quokka.js Console Output

_See the description above._

### Code editor version

Visual Studio Code v1.84.2

### OS name and version

OSX Sonoma (14.1.2)

**Comment by OkiMusix05** [comment]
Having the same issue. Using React and Vite and I want to read the contents of a file. This is my code:
`const path = url;`
`const file = Bun.file(path);`
`const text = await file.text();`
and it doesn't work. It gives me "Bun is not defined" error.
I don't know if this is relevant, but I'm using OSX Monterey.

**Comment by Velociraptor115** [comment]
I would also like to be able to use the Quokka extension with Bun

My primary use of Bun is for smaller scripts related to a bigger typescript repo without going through the ts transpilation step
If quokka can support bun as an alternative runner, it would be great

**Comment by markusstrasser** [comment]
Yes, please add bun support. Realtime values logging would be especially useful since bun natively also can do .js, .ts., .tsx and so on

**Comment by flexchar** [comment]
I bought licence as well and didn't check that Bun is not yet supported. I hope it'd be easy since it provides `Bun -e` to evaluate the code. Super nice!

**Comment by DHFW** [comment]
Please!

**Comment by smcenlly** [comment]
Quokka now supports running your code using [Bun](https://bun.sh/) as its runtime.

If you already use Bun in one or more of your project.json scripts, Quokka will prompt you to use Bun to run your code. 

You may also configure Quokka to always use Bun by default in your [Quokka settings](https://quokkajs.com/docs/config/runtime/#bun-configuration).
